crates/ra_lsp_server/src/markdown.rs

```diff
@@ -0,0 +1,38 @@
+use ra_ide_api::Documentation;
+
+pub(crate) fn sanitize_markdown(docs: Documentation) -> Documentation {
+    let docs: String = docs.into();
+
+    // Massage markdown
+    let mut processed_lines = Vec::new();
+    let mut in_code_block = false;
+    for line in docs.lines() {
+        if line.starts_with("```") {
+            in_code_block = !in_code_block;
+        }
+
+        let line = if in_code_block && line.starts_with("```") && !line.contains("rust") {
+            "```rust".into()
```

Comment:
This will turn ```` ```plain ```` into ```` ```rust ````.

Reply:
I woudn't be against pulling-in a markdown parser...

Reply:
Agreed. I think ultimately we could do things like: text only clients get stripped out "```" for readability, and we could run doc tests with a code-lens etc. We'll probably also need one to make links work in markdown.